src/check_names.cpp

```diff
@@ -0,0 +1,76 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// jsonify a character vector
+std::string jsonify(CharacterVector x) {
```

Comment:
Since this is hopefully not going to be called very often (if at all), it's not performance sensitive. It may be better to use base::paste instead, like:

    Function paste = Environment("package:base")["paste"];
    return paste(x, Named("collapse", ", "));